sim/hunter/hunter.go

```diff
@@ -139,53 +149,34 @@ func NewHunter(character *core.Character, options *proto.Player, hunterOptions *
 		Options:   hunterOptions,
 	}
 	core.FillTalentsProto(hunter.Talents.ProtoReflect(), options.TalentsString, TalentTreeSizes)
-	hunter.EnableManaBar()
+
+	// Todo: Verify that is is actually 4 focus per second
+	hunter.EnableFocusBar(100 + (float64(hunter.Talents.KindredSpirits) * 5), 4.0, true)
 
 	hunter.PseudoStats.CanParry = true
 
-	rangedWeapon := hunter.WeaponFromRanged(0)
 
 	// Passive bonus (used to be from quiver).
-	hunter.PseudoStats.RangedSpeedMultiplier *= 1.15
-
-	if hunter.HasRangedWeapon() && hunter.GetRangedWeapon().ID != ThoridalTheStarsFuryItemID {
-		switch hunter.Options.Ammo {
-		case proto.HunterOptions_IcebladeArrow:
-			hunter.AmmoDPS = 91.5
-		case proto.HunterOptions_SaroniteRazorheads:
-			hunter.AmmoDPS = 67.5
-		case proto.HunterOptions_TerrorshaftArrow:
-			hunter.AmmoDPS = 46.5
-		case proto.HunterOptions_TimelessArrow:
-			hunter.AmmoDPS = 53
-		case proto.HunterOptions_MysteriousArrow:
-			hunter.AmmoDPS = 46.5
-		case proto.HunterOptions_AdamantiteStinger:
-			hunter.AmmoDPS = 43
-		case proto.HunterOptions_BlackflightArrow:
-			hunter.AmmoDPS = 32
-		}
-		hunter.AmmoDamageBonus = hunter.AmmoDPS * rangedWeapon.SwingSpeed
-		hunter.NormalizedAmmoDamageBonus = hunter.AmmoDPS * 2.8
+	//hunter.PseudoStats.RangedSpeedMultiplier *= 1.15
+	rangedWeapon := hunter.WeaponFromRanged(0)
+
+	hunter.EnableAutoAttacks(hunter, core.AutoAttackOptions{
+		// We don't know crit multiplier until later when we see the target so just
+		// use 0 for now.
+		MainHand:        hunter.WeaponFromMainHand(0),
+		OffHand:         hunter.WeaponFromOffHand(0),
+		Ranged:          rangedWeapon,
+		//ReplaceMHSwing:  hunter.TryRaptorStrike, //Todo: Might be weaving
+		AutoSwingRanged: true,
+	})
+
+	hunter.AutoAttacks.RangedConfig().ApplyEffects = func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+		baseDamage := hunter.RangedWeaponDamage(sim, spell.RangedAttackPower(target)) +
+			spell.BonusWeaponDamage()
+		spell.CalcAndDealDamage(sim, target, baseDamage, spell.OutcomeRangedHitAndCrit)
```

Comment:
do you need to do this if you're not changing it from the default implementation? Wotlk/Sod do this to add the Ammo bonus damage which seems to be gone now

Reply:
Not sure how the default implementation works. but I removed it for now, the outcome seems to be the same.